File: src/main/java/cn/labzen/logger/spring/ls/LabzenLoggingSystem.java
package cn.labzen.logger.spring.ls;

import org.springframework.boot.logging.AbstractLoggingSystem;
import org.springframework.boot.logging.LogFile;
import org.springframework.boot.logging.LoggingInitializationContext;

public abstract class LabzenLoggingSystem extends AbstractLoggingSystem {

  public LabzenLoggingSystem(ClassLoader classLoader) {
    super(classLoader);
  }

  @Override
  protected void loadDefaults(LoggingInitializationContext initializationContext, LogFile logFile) {
    // do nothing
  }

  @Override
  protected void loadConfiguration(LoggingInitializationContext initializationContext,
                                   String location,
                                   LogFile logFile) {
    if (initializationContext != null) {
      applySystemProperties(initializationContext.getEnvironment(), logFile);
    }
  }
}

File: src/main/kotlin/cn/labzen/logger/spring/ls/LabzenLoggingSystem.kt
package cn.labzen.logger.spring.ls

import org.springframework.boot.logging.AbstractLoggingSystem
import org.springframework.boot.logging.LogFile
import org.springframework.boot.logging.LoggingInitializationContext

abstract class LabzenLoggingSystem(classLoader: ClassLoader) : AbstractLoggingSystem(classLoader) {

  override fun loadDefaults(initializationContext: LoggingInitializationContext, logFile: LogFile?) {
    // do nothing
  }

  override fun loadConfiguration(
    initializationContext: LoggingInitializationContext?,
    location: String,
    logFile: LogFile?
  ) {
    if (initializationContext != null) {
      applySystemProperties(initializationContext.environment, logFile)
    }
  }
}
